providers/oidc.go

```diff
@@ -279,7 +294,24 @@ func (p *OIDCProvider) findClaimsFromIDToken(ctx context.Context, idToken *oidc.
 type OIDCClaims struct {
 	rawClaims         map[string]interface{}
 	UserID            string
-	Subject           string `json:"sub"`
-	Verified          *bool  `json:"email_verified"`
-	PreferredUsername string `json:"preferred_username"`
+	Subject           string   `json:"sub"`
+	Verified          *bool    `json:"email_verified"`
+	PreferredUsername string   `json:"preferred_username"`
+	Groups            []string `json:"groups"`
+}
+
+func (c *OIDCClaims) containsGroups(groups []string) bool {
+	m := map[string]bool{}
+
+	for _, g := range groups {
```

Comment:
Nice! Thanks for keeping this O(N) and not nesting the for loops 👍

Reply:
I just realized the logic here is wrong, it will return true when any group exists, needs to be ALL, fixing up as part of the ValidateGroup change.

Reply:
I disagree (I had a thread about this somewhere else, I can try to find the details) -- and think if a user is a member of one of the groups, they should be allowed (this is how the other providers that go group authorization behave I believe).

IE a very likely corporate structure, I have SREs, Devs & Designers each in a group of those names.  I want to allow them all to a route, so I would want the group authorization to allow someone in any of those 3 groups.

Reply:
Ok that makes sense then, I was more originally thinking of it as an AND but implemented it wrong, but turns out it was perhaps right :laughing: